examples/failing/UnusableTypeClassMethodSynonym.purs

```diff
@@ -0,0 +1,9 @@
+-- @shouldFailWith UnusableDeclaration
+module Main where
+
+type M x = forall a. a
+
+class C a where
+  -- after synonym expansion, the type doesn't actually contain an `a`
+  c :: M a
```

Comment:
Have I misunderstood what `freeTypeVariables` does?  I was expecting to not get back any explicitly bound variables, such as those in introduced in a forall.